extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java

```diff
@@ -0,0 +1,107 @@
+package org.eclipse.dataspaceconnector.azure.testfixtures;
+
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.List;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import static java.io.File.separator;
+import static java.lang.String.format;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+
+public interface CosmosTestClient {
+
+    static CosmosClient createClient() {
+        var cosmosKey = propOrEnv("COSMOS_KEY", null);
+        if (cosmosKey != null) {
+            return clientForCi(cosmosKey);
+        } else {
+            return localClient();
+        }
+    }
+
+    private static CosmosClient clientForCi(String cosmosKey) {
+        return new CosmosClientBuilder()
+                .key(cosmosKey)
+                .preferredRegions(List.of("westeurope"))
+                .endpoint("https://cosmos-itest.documents.azure.com:443/")
+                .buildClient();
+    }
+
+    private static CosmosClient localClient() {
+        try {
+            var endpoint = "https://127.0.0.1:8081";
+
+            var keystorePath = System.getProperty("java.home") + separator + "lib" + separator + "security" + separator + "cacerts";
+            var keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+
+            var keystorePassword = "changeit".toCharArray();
+            keystore.load(new FileInputStream(keystorePath), keystorePassword);
+
+            var certificate = getCertificateFrom(endpoint);
+            keystore.setCertificateEntry("cosmosdb", certificate);
+
+            try (var output = new FileOutputStream(keystorePath)) {
```

Comment:
I get this. Would need to run as root.

```
Caused by:
        java.io.FileNotFoundException: /usr/lib/jvm/java-11-openjdk-amd64/lib/security/cacerts (Permission denied)
            at java.base/java.io.FileOutputStream.open0(Native Method)
            at java.base/java.io.FileOutputStream.open(FileOutputStream.java:298)
            at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:237)
            at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:126)
            at org.eclipse.dataspaceconnector.azure.testfixtures.CosmosTestClient.localClient(CosmosTestClient.java:58)
            ... 2 more
```

Reply:
This seems to work though

```

            var trustStorePath = someTempFile;
            try (var output = new FileOutputStream(trustStorePath)) {
                keystore.store(output, keystorePassword);
            }
            System.setProperty("javax.net.ssl.trustStore", trustStorePath);
```

Reply:
Good catch, I use sdkman so my jdk is in the user folder and I have no problems with permissions.